src/resolve/resolved-dns-stream.c

```diff
@@ -183,6 +183,36 @@ static int dns_stream_identify(DnsStream *s) {
         return 0;
 }
 
+static ssize_t dns_stream_writev(DnsStream *s, struct iovec *iov, int iovcnt) {
+        ssize_t r;
+
+        assert(s);
+        assert(iov);
+
+        if (s->tfo_salen) {
+                struct msghdr hdr = {};
+
+                hdr.msg_iov = iov;
+                hdr.msg_iovlen = iovcnt;
+                hdr.msg_name = &s->tfo_address->sa;
+                hdr.msg_namelen = s->tfo_salen;
+                r = sendmsg(s->fd, &hdr, MSG_FASTOPEN);
+                if (r < 0) {
+                        if (errno == EOPNOTSUPP) {
+                                mfree(s->tfo_address);
```

Comment:
mfree() is a wrapper around free() with the single purpose of also returning NULL, so that you can write stuff like:

```c
p = mfree(p);
```

but, if you don't assign the result to anything, then there's no point in using it, you might as well use regular free()